### PR TITLE
Issue 999  random state

### PR DESCRIPTION
**Pull request by adler-j**

### CODE REVIEW

File: odl/util/utility.py [hunk 1]
```diff
@@ -805,6 +805,43 @@ def pkg_supports(feature, pkg_version, pkg_feat_dict):
     return False
 
 
+class NumpyRandomSeed(object):
+    """Context manager for numpy randoms seeds."""
```
Comment (kohr-h) [inline odl/util/utility.py]: Numpy random seeds

File: odl/util/utility.py [hunk 2]
```diff
@@ -805,6 +805,43 @@ def pkg_supports(feature, pkg_version, pkg_feat_dict):
...
+
+    def __init__(self, seed):
+        """
```
Comment (kohr-h) [inline odl/util/utility.py]: Missing

File: odl/util/utility.py [hunk 3]
```diff
@@ -805,6 +805,43 @@ def pkg_supports(feature, pkg_version, pkg_feat_dict):
...
+
+        Parameters
+        ----------
+        seed : int or None
+            Seed value for the random number generator.
+            ``None`` is interpreted as keeping the current seed.
+
+        Examples
+        --------
+        Always returns same value with same seed:
```
Comment (kohr-h) [inline odl/util/utility.py]: Make text a bit nicer, e.g., "Use this to make drawing pseudo-random numbers repeatable:"

File: odl/phantom/noise.py [hunk 4]
```diff
@@ -125,6 +140,9 @@ def salt_pepper_noise(vector, fraction=0.05, salt_vs_pepper=0.5,
         The "salt" value in the noise.
         Default: maximuim value of ``vector``. For product spaces the maximum
         value per subspace is taken.
+    seed : int, optional
+        Random seed to use for generating the noise. If ``None``, uses current
```
Comment (kohr-h) [inline odl/phantom/noise.py]: For \`\`None\`\`, use the current seed

File: odl/util/utility.py (hunk 2 above)
Comment (adler-j) [inline odl/util/utility.py]: What do we write here? "Initialize an instance"?

Comment (kohr-h) [inline odl/util/utility.py]: Why not? You can as well create an instance of the context manager and then write `with instance:`.

**Pull request merged**